### ISSUE TITLE
Replacing " with ' causes two ' to be added at the end of a word in JavaScript files

### ISSUE THREAD

**Issue by wagerfield** [Microsoft/vscode#12239]
- VSCode Version: 1.5.2
- OS Version: 10.11.6

Steps to Reproduce:
1. Create a JavaScript file and write `var foo = 'foo'`
2. Select one of the single quote marks and then hit `CMD+D` to have VS Code select the other one
3. With both single quote marks selected hit `DELETE` to remove them
4. Hit the single _or_ double quote mark key to add the quote marks back in
5. Observe that two quote marks are added at the end of the word

This is unexpected behaviour and not inline with what SublimeText does. When pasting code from the net using double quote marks, I select one of the quote marks, hold `CMD+D` until all of them are selected, then hit `DELETE`, then hit the single quote key. At least this is what I used to do in SublimeText.

I appreciate that what I am trying to achieve can be done with 'find and replace', but this is a habit I have formed that is faster and more familiar to me...don't judge :)

**Comment by xavadu** [comment]
I was looking for it,

If a single quote or double quote is added at the end of a word it should not auto add the closing one.
Usually on that case, a user is closing a quote, not trying to add something between quotes.

```
// Having the following:
"test

// Quote typed at the end (and automatic extra closing quote is added) 
"test""

// Desire behavior (not automatic extra closing quote is added when a quite is typed at  the end )
"test"
```

As described in the main message, it also happen if you want to replace it by selection or multi-selection.

The actual behavior at the moment is very annoying.

**Comment by alexandrudima** [comment]
Fixed via 19a881b72fbdd0b34cb769bc978d3fbadd6a40de -- now all cursors "do the same thing" (at least when typing)
